.ci/unit-tests/Robot_Adapter_Tests/PullTests.cs

```diff
@@ -0,0 +1,113 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using NUnit.Framework;
+using BH.oM.Base;
+using BH.oM.Structure.Constraints;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.Loads;
+using BH.oM.Structure.MaterialFragments;
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Structure.SurfaceProperties;
+using System.Diagnostics.Contracts;
+using BH.Adapter.Robot;
+using BH.oM.Data.Requests;
+using BH.oM.Adapters.Robot;
+using BH.oM.Analytical.Results;
+using System;
+using Shouldly;
+using BH.oM.Adapter.Commands;
+using System.Reflection;
+
+namespace BH.Tests.Adapter.Robot
+{
+    public class PullTests
+    {
+        RobotAdapter m_Adapter;
+        [OneTimeSetUp]
+        public void OneTimeSetup()
+        {
+            //Starts the RobotAdapter and opens the model file stored in the .ci folder
+            string currentDirectory = AppDomain.CurrentDomain.BaseDirectory;
+            List<string> splitPath = currentDirectory.Split('\\').ToList();
+            splitPath = splitPath.Take(splitPath.IndexOf(".ci") + 2).ToList();
+            string modelPath = Path.Join(string.Join("\\", splitPath), "Models", "Simple 2-story structure with results.rtd");
+            m_Adapter = new RobotAdapter(modelPath, null, true);
+            //Forces Analytical_Engine to be loaded up. This is due to some parts of the call-chain not hard wired. This ensures that methods from Analytical_Engine are loaded and usable from RunExtensionMethod
+            Assembly.Load("Analytical_Engine");
```

Comment:
Just leaving a comment here to record our useful chat around this! 😄 

The issue is that because we leverage Reflection to call certain methods (the Extension methods, via `TryRunExtensionMethod()`), the Unit Test program fails to identify that certain assemblies are needed in memory. This happens regardless of whether such needed assemblies are referenced within the project dependencies or not. This behaviour is due to the .NET CLR Just-In-Time compilation. Among other things, this means it loads assemblies on first use. So, despite assemblies being referenced by an assembly in use, if the references haven't yet been needed by the CLR to execute the program, they're not loaded and so will not appear in the list of assemblies in the current AppDomain.

In practice, this means that we have a couple of option to make the test work in such cases:
1. Have a mechanism that is run once per Unit Test setup ("OneTimeSetup") which makes sure that all assemblies referenced in the Unit Test project are loaded. This could be done by calling `Assembly.GetReferencedAssemblies()` and calling `Assembly.Load()` on all of them. This would have negligible impact because if some of the ref assemblies are already loaded, they will not be loaded again. It still retains the advantage of not loading _all_ assemblies in the BHoM Assemblies folder, which it demonstrably takes too long to be compatible with the nature of TDD.
2. The alternative is what has been done here: manually loading the assemblies that are known to miss. This option has a negligible speed advantage over option 1 (you only skip the call to `Assembly.GetReferencedAssemblies()` and the subsequent loop) and it has a major drawback. How does a developer become aware that some assemblies are not loaded, and that this missing load is responsible for an incorrect behavior? Best scenario an exception is thrown during test run. However, worst case is that some _weird_ behavior arises during testing, which is only reported via warnings or, worse, not at all (e.g. because a fallback condition kicks in). It can be hard to identify the missing load issue for a person less familiar with BHoM. This latter scenario is what happened here, and it took @IsakNaslundBh and me some time to understand and identify that we needed to explicitly load `Analytical_Engine`: there were no exceptions, just some telling behavior and messages passed around.

Reply:
Following the previous, I'm proposing a new base `NUnitTest` class to be created in the Test_Toolkit, which can implement a `OneTimeSetup` method with the content of point 1 above. This class is to be inherited by any other NUnit testing class in downstream solutions. The inheritance could be the target of compliance checks. 
This has been discussed with CI/CD @FraserGreenroyd